lib/vdsm/virt/migration.py

```diff
@@ -648,11 +650,38 @@ def _migration_params(self, muri):
                     continue
                 cputune.remove(vcpu)
         if self._destination_cpusets is not None:
+            # Reconfigure CPU pinning based on call parameter
             if cputune is None:
                 cputune = xml.etree.ElementTree.Element('cputune')
                 dom.append(cputune)
-            for vcpupin in self._destination_cpusets:
+            # First modify existing elements
+            for vcpupin in vmxml.find_all(cputune, 'vcpupin'):
+                vcpu_id = int(vcpupin.get('vcpu'))
+                if vcpu_id > 0 and vcpu_id < len(self._destination_cpusets):
```

Comment:
Really vcpu_id > 0 and not vcpu_id >= 0?

And is it expected to see vcpu_id outside the given range or should a warning be logged in such a case?

Reply:
> Really vcpu_id > 0 and not vcpu_id >= 0?

Yes, it should be `>= 0`. I have it fixed locally just didn't push it yet.


> And is it expected to see vcpu_id outside the given range or should a warning be logged in such a case?

It is possible to see such `vcpu_id`. There may be vCPU pinning for vCPUs that are defined but not active (ready for hot-plugging). We don't really care about those here (correct pinning will be specified during hot-plug). Producing a warning sounds superfluous. We could go with info level, but I still think this is not necessary.

Reply:
I see, sure, no log message needed in such a case.